src/utils.js

```diff
@@ -23,3 +25,23 @@ module.exports.maybeCallback = (promise, callback) => {
 
   return promise;
 };
+
+module.exports.addLegacyMetadata = options => {
+  if (options.driverInfo == null) {
```

Comment:
I guess we don't have access to node 14 `?` and `??` operators here, but I think we can still condense this logic to be a bit easier to parse, and we should also improve this to make sure we defer error handling to the base class as much as we can in the case that driverInfo isn't a proper object, too. Something along these lines (you can stick to template strings if you really want, but the idea is the same).

```js
  if (typeof options.driverInfo !== 'object') {
    return;
  }
  options.driverInfo = options.driverInfo || {};
  const infoParts = {
    name: ['mongodb-legacy'],
    version: [version]
  };
  for (const prop of ['name', 'version']) {
    if (typeof options.driverInfo[prop] === 'string') {
      infoParts[prop].push(options.driverInfo[prop]);
    }
    options.driverInfo[prop] = infoParts[prop].join('|');
  }
```